File: src/main/java/inlinekeyword/java/InlineKeyword.java
package inlinekeyword.java;

import java.util.Map;
import java.util.Properties;
import org.jetbrains.annotations.NotNull;

@SuppressWarnings("ALL")
public final class InlineKeyword {
    @NotNull
    private static final Properties properties = new Properties();

    public static final void main() {
        double radius = 5.5;
        String string = "A circle with radius " + radius + " has a circumference of " + circumference(radius);
        System.out.println(string);
    }

    @NotNull
    public static final Properties getProperties() {
        return properties;
    }

    public static final double circumference(double radius) {
        double d = Math.PI * 2 * radius;
        double c = d;
        Properties properties = getProperties();
        Map map = properties;
        String string = "previousToLast";
        Object object = properties.get("last");
        map.put(string, object);
        string = "last";
        object = c;
        map.put(string, object);
        return d;
    }

    public static /* synthetic */ void main(String[] args) {
        main();
    }
}

File: src/main/kotlin/inlinekeyword/kotlin/InlineKeyword.kt
package inlinekeyword.kotlin

import java.util.*
import kotlin.math.PI

fun main() {
    val radius = 5.5
    println("A circle with radius $radius has a circumference of ${circumference(radius)}")
}

val properties = Properties()

fun circumference(radius: Double): Double {
    return (2 * PI * radius).also { c ->
        properties.apply {
            set("previousToLast", get("last"))
            set("last", c)
        }
    }
}
